hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceSlowTest.java

```diff
@@ -103,23 +103,26 @@ public void stats_longRunningTask_durable()
 
         String key = generateKeyOwnedBy(instances[1]);
 
-        ICountDownLatch latch = instances[0].getCountDownLatch("latch");
-        latch.trySetCount(6);
+        ICountDownLatch firstLatch = instances[0].getCountDownLatch("firstLatch");
+        firstLatch.trySetCount(2);
+
+        ICountDownLatch lastLatch = instances[0].getCountDownLatch("lastLatch");
+        lastLatch.trySetCount(6);
 
         IScheduledExecutorService executorService = getScheduledExecutor(instances, "s");
         IScheduledFuture future = executorService.scheduleOnKeyOwnerAtFixedRate(
-                new ICountdownLatchRunnableTask("latch"), key, 0, 10, SECONDS);
+                new ICountdownLatchRunnableTask("firstLatch", "lastLatch"), key, 0, 10, SECONDS);
 
-        Thread.sleep(12000);
+        firstLatch.await(12, SECONDS);
 
         instances[1].getLifecycleService().shutdown();
 
-        latch.await(70, SECONDS);
+        lastLatch.await(70, SECONDS);
         sleepSeconds(4); // Wait for run-cycle to finish before cancelling, in order for stats to get updated.
         future.cancel(false);
 
         ScheduledTaskStatistics stats = future.getStats();
-        assertEquals(6, stats.getTotalRuns());
+        assertEquals(6, stats.getTotalRuns(), 1);
```

Comment:
@vbekiaris thanks for your time on providing an approach. That was the obvious approach, but as I said above, every runnable is executed by https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.java#L375 which will wrap the run, and no matter what you do in it to prevent stats updating, it will update them through the finally block. Cancelling the Future which you nicely did, is not a deterministic action, because we can not guarantee interruption of the task, thus, the stats may or not, get updated. And in your case, where counter == executions, we do want them to get updated, so if you successfully cancel it, (again not deterministic) then the test will fail, with countdownLatch = 5. Last, keep in mind that the scheduled tasks runs _outside_ the partition threads.